slack_sdk/oauth/installation_store/amazon_s3/__init__.py

```diff
@@ -282,15 +283,52 @@ def delete_installation(
             Bucket=self.bucket_name,
             Prefix=f"{workspace_path}/installer-{user_id or ''}",
         )
+        deleted_keys = []
         for content in objects.get("Contents", []):
             key = content.get("Key")
             if key is not None:
                 self.logger.info(f"Going to delete installation ({key})")
                 try:
                     self.s3_client.delete_object(
                         Bucket=self.bucket_name,
-                        Key=content.get("Key"),
+                        Key=key,
                     )
+                    deleted_keys.append(key)
+                except Exception as e:  # skipcq: PYL-W0703
+                    message = f"Failed to find bot installation data for enterprise: {e_id}, team: {t_id}: {e}"
+                    raise SlackClientConfigurationError(message)
+
+                try:
+                    no_user_id_key = key.replace(f"-{user_id}", "")
```

Comment:
When saving a new installation, this implementation stores two historical data: 
`/111.222/E111-T111/installer-1627859348.984687` and `/111.222/E111-T111/installer-U111-1627859348.984687` to support both `find_installation(enterprise_id, team_id)` and `find_installation(enterprise_id, team_id, user_id)` without performance overhead (= scanning all the keys under the directory). This deletion method used to delete only the first one.